src/otx/algorithms/common/adapters/mmcv/utils/config_utils.py

```diff
@@ -853,37 +871,109 @@ def _set_size_value(pipeline: Dict, attr: str, scale: Tuple[Union[int, float], U
-) -> Union[None, Tuple[int, int]]:
-    """Get configurable input size configuration. If it doesn't exist, return None.
+        Args:
+            input_size_config (InputSizePreset, optional): Input size setting. Defaults to InputSizePreset.DEFAULT.
+            model_ckpt (Optional[str], optional): Model weight to load. Defaults to None.
 
-    Args:
-        input_size_config (InputSizePreset, optional): Input size configuration. Defaults to InputSizePreset.DEFAULT.
-        model_ckpt (Optional[str], optional): Model weight to load. Defaults to None.
+        Returns:
+            Optional[Tuple[int, int]]: Pair of width and height. If there is no input size configuration, return None.
+        """
+        input_size = None
+        if input_size_config == InputSizePreset.DEFAULT:
+            if model_ckpt is None:
+                return None
 
-    Returns:
-        Union[None, Tuple[int, int]]: Pair of width and height. If there is no input size configuration, return None.
-    """
-    input_size = None
-    if input_size_config == InputSizePreset.DEFAULT:
-        if model_ckpt is None:
-            return None
-
-        model_info = torch.load(model_ckpt, map_location="cpu")
-        for key in ["config", "learning_parameters", "input_size", "value"]:
-            if key not in model_info:
+            model_info = torch.load(model_ckpt, map_location="cpu")
+            for key in ["config", "learning_parameters", "input_size", "value"]:
+                if key not in model_info:
+                    return None
+                model_info = model_info[key]
+            input_size = model_info
+
+            if input_size == InputSizePreset.DEFAULT.value:
                 return None
-            model_info = model_info[key]
-        input_size = model_info
 
-        if input_size == InputSizePreset.DEFAULT.value:
-            return None
+            logger.info("Given model weight was trained with {} input size.".format(input_size))
 
-        logger.info("Given model weight was trained with {} input size.".format(input_size))
-    else:
-        input_size = input_size_config.value
+        else:
+            input_size = input_size_config.value
+
+        return InputSizePreset.parse(input_size)
+
+    @staticmethod
+    def select_closest_size(input_size: Tuple[int, int], preset_sizes: List[Tuple[int, int]]):
+        """Select the most closest size from preset sizes in log scale.
 
-    parsed_tocken = re.match("(\\d+)x(\\d+)", input_size)
-    return (int(parsed_tocken.group(1)), int(parsed_tocken.group(2)))
+        Args:
+            input_size (Tuple[int, int]): Query input size
+            preset_sizes (List[Tuple[int, int]]): List of preset input sizes
+
+        Returns:
+            Tuple[int, int]: Best matching size out of preset. Returns input_size if preset is empty.
+        """
+        if len(preset_sizes) == 0:
+            return input_size
+
+        def scale_of(x):
```

Comment:
```suggestion
        def to_log_scale(x):
```
This might be more clear

Reply:
Will fix in next commit.